bin/user/weatherservices.py

```diff
@@ -1655,6 +2221,9 @@ def _to_weewx(self, thread_name, reply, usUnits):
             else:
                 try:
                     val = reply[key]
+                    # first convert val to standard unit in METRIC unit system
+                    val = weewx.units.convertStd(val, weewx.METRIC)
+                    # now convert val to archive record unit system
```

Comment:
Warum das? Der konvertiert von jeder zu jeder Einheit. Wenn man Einheiten anlegt, muß man übrigens grundsätzlich auch alle Umrechnungen zwischen der neuen Einheit und den vorhandenen Einheiten eintragen. Man kann nie vorhersehen, wo irgendetwas hin- und herkonvertiert wird.

Reply:
Ich hatte ein Problem, dass die metrische Einheit nicht in die US Einheit konvertiert werden konnte, da es keine Konvertierungsfunktion gab. Ich weiß nicht mehr genau was es war, evtl. 'km' zu 'foot' oder so. Daher habe ich erst die Konvertierung in die Standardeinheit durchgeführt und dann zum Archiv-Record Unit System. Mir fehlt da einfach zu viel um das vernünftig zu machen. Du hast jetzt mehr Arbeit das durchzusehen als wenn Du es gleich richtig gemacht hättest.